snappass/main.py

```diff
@@ -99,6 +100,8 @@ def handle_password():
 
 @app.route('/<password_key>', methods=['GET'])
 def show_password(password_key):
+    if re.search('bot', request.headers.get('User-Agent', '')):
```

Comment:
This is a good idea. Do you think we could move this out to a function called `request_is_valid(request)`, that returns a boolean? That way in the future we may want to append some logic in it.